Bottom: ca7e9cfe2c5c9c76c2477569047c8430973713a9
Top:    2a7c9bb5600112f68a6f3f8ed162e6f985ca8a4b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-10 00:45:52 +0000

Refresh of hid-api-make-hid-prompt_for-re

---

diff --git a/src/action.c b/src/action.c
index 9e1370b..07b2752 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1408,6 +1408,8 @@ NotifyMode (void)
 		    Draw ();
 		  }
 		}
+
+	    /* free memory allocated by gui->prompt_for() */
 	    free (string);
 	  }
 	break;
@@ -4649,8 +4651,7 @@ ActionChangeName (int argc, char **argv, int x, int y)
 	case F_Layout:
 	  name =
 	    gui->prompt_for (_("Enter the layout name:"), EMPTY (PCB->Name));
-	  /* NB: ChangeLayoutName takes ownership of the passed memory */
-	  if (name && ChangeLayoutName (name))
+	  if (name && ChangeLayoutName (name))	/* XXX memory leak */
 	    SetChangedFlag (true);
 	  break;
 
@@ -4658,8 +4659,7 @@ ActionChangeName (int argc, char **argv, int x, int y)
 	case F_Layer:
 	  name = gui->prompt_for (_("Enter the layer name:"),
 				  EMPTY (CURRENT->Name));
-	  /* NB: ChangeLayerName takes ownership of the passed memory */
-	  if (name && ChangeLayerName (CURRENT, name))
+	  if (name && ChangeLayerName (CURRENT, name))	/* XXX memory leak */
 	    SetChangedFlag (true);
 	  break;
 	}
@@ -5951,8 +5951,7 @@ ActionNew (int argc, char **argv, int x, int y)
       CreateNewPCBPost (PCB, 1);
 
       /* setup the new name and reset some values to default */
-      free (PCB->Name);
-      PCB->Name = name;
+      PCB->Name = name;		/* XXX memory leak */
 
       ResetStackAndVisibility ();
       CreateDefaultFont ();
@@ -7606,7 +7605,7 @@ ActionImport (int argc, char **argv, int x, int y)
 
   if (mode && strcasecmp (mode, "setdisperse") == 0)
     {
-      char *ds, *units;
+      const char *ds, *units;
       char buf[50];
 
       ds = ARG (1);
@@ -7623,8 +7622,6 @@ ActionImport (int argc, char **argv, int x, int y)
 	}
       else
 	AttributePut (PCB, "import::disperse", ds);
-      if (ARG (1) == NULL)
-        free (ds);
       return 0;
     }
 
diff --git a/src/change.c b/src/change.c
index 2545d84..df7f454 100644
--- a/src/change.c
+++ b/src/change.c
@@ -1079,7 +1079,6 @@ ChangeTextName (LayerTypePtr Layer, TextTypePtr Text)
 bool
 ChangeLayoutName (char *Name)
 {
-  free (PCB->Name);
   PCB->Name = Name;
   hid_action ("PCBChanged");
   return (true);
@@ -1107,7 +1106,6 @@ ChangeElementSide (ElementTypePtr Element, LocationType yoff)
 bool
 ChangeLayerName (LayerTypePtr Layer, char *Name)
 {
-  free (CURRENT->Name);
   CURRENT->Name = Name;
   hid_action ("LayersChanged");
   return (true);
@@ -2282,7 +2280,7 @@ QueryInputAndChangeObjectName (int Type, void *Ptr1, void *Ptr2, void *Ptr3)
     }
   if (name)
     {
-      /* NB: ChangeObjectName takes ownership of the passed memory */
+      /* XXX Memory leak!! */
       char *old = ChangeObjectName (Type, Ptr1, Ptr2, Ptr3, name);
       if (old != (char *) -1)
 	{
